### PR TITLE
Refactor spring boot starters

### PR DESCRIPTION
**Pull request by kpavlov**
Refactor LLM client auto-configuration structure

- Removed `KoogAutoConfiguration` and related legacy configuration files.
- Introduced modular `LLM` provider-specific auto-configuration classes (e.g., `OpenAILLMAutoConfiguration`).
- Updated test setup to align with the new modular configuration.
- Enhanced property validation with `enabled` flag support for finer-grained control.
- Introduced a `String.masked` extension to obfuscate sensitive string parts. Added test coverage for various input scenarios, including custom mask characters and edge cases.
- Updated `KoogAutoConfigurationTest` to include scenarios where providers are explicitly disabled.
- Refactored parameterized tests to use `textBlock = PROVIDERS` with `@CsvSource`.
- Adjusted `ApplicationContextRunner` configuration logic for clean test scenarios.
- Introduced `ignoreUnknownFields` to all `ConfigurationProperties` to improve flexibility with unmapped fields.
- Expanded property documentation for all LLM clients (e.g., OpenAI, Google, Anthropic) to include detailed descriptions, usage examples, and retry behavior.
- Improved autowiring safety with `@ConditionalOnBean` for LLM-specific executors.
- Updated `spring-boot.md` to detail `enabled` flags and their default behavior for LLM providers.
- Improved clarity by reformatting the supported provider list.
- Added explanations for default base URLs and explicit activation for non-API-key providers (e.g., Ollama).

A blog post about writing Spring Boot starters is [here](https://kpavlov.me/blog/spring-boot-starters) 

**NB! Tests need a split and some love, but this PR is already big enough. I will refactor them separately.**

## Motivation and Context

- The Spring Boot configuration in general is not very Spring Boot idiomatic
- The current Spring Boot starter does not expose LLM Clients. In order to get the LLM client in the application, developers have to use reflection to a get a field from the executor 🤯 
- All configuration is encapsulated in one file, which makes extensibility difficult (SRP is violated).
- Defaults are hardcoded in classes instead of being provided in the resources
- Ollama is enabled by the presence of the base URL, which is interesting.
- There is no explicit way to disable an LLM provider: no `enabled=true` property

## Breaking Changes

- AutoConfiguration classes moved to their respective packages
- Ollama provider needs `ai.koog.ollama.enabled=true` to be set explicitly in order to activate.

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [x] Tests improvement
- [x] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [ ] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [ ] Tests for the changes have been added
- [ ] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [ ] An issue describing the proposed change exists
- [ ] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [x] Docs have been added / updated

### CODE REVIEW

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import ai.koog.spring.RetryConfigKoogProperties
+
+/**
+ * Interface representing configuration properties for a Koog LLM Client.
+ *
+ * This interface is intended to provide the necessary configuration required to set up a LLM Client.
+ * It includes options for enabling the client, specifying the base URL, and defining retry configurations.
+ *
+ * @param enabled Indicates whether the LLM client is enabled.
```
Comment (devcrocod) [inline koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt:11]: It seems better to change it to `property` so that the documentation is available on the properties.

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/anthropic/AnthropicKoogProperties.kt
Comment (devcrocod) [inline koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/anthropic/AnthropicKoogProperties.kt:1]: Why name the packages like that?
Maybe it’s better to just use `ai.koog.spring.clients.*`

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/openai/OpenAILLMAutoConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    public fun openAILLMClient(): OpenAILLMClient {
+        return OpenAILLMClient(
+            apiKey = properties.apiKey,
+            settings = OpenAIClientSettings(baseUrl = properties.baseUrl)
+        )
+    }
+
+    /**
+     * Provides a [SingleLLMPromptExecutor] bean configured with a [GoogleLLMClient] using the settings
+     * from the given `KoogProperties`. The bean is only created if the `google.api-key` property is set.
```
Comment (devcrocod) [inline koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/openai/OpenAILLMAutoConfiguration.kt]: `google.api-key` -> `openai.api-key`

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/openai/OpenAILLMAutoConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    @Bean
+    public fun openAILLMClient(): OpenAILLMClient {
+        return OpenAILLMClient(
+            apiKey = properties.apiKey,
+            settings = OpenAIClientSettings(baseUrl = properties.baseUrl)
+        )
+    }
+
+    /**
+     * Provides a [SingleLLMPromptExecutor] bean configured with a [GoogleLLMClient] using the settings
```
Comment (devcrocod) [inline koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/openai/OpenAILLMAutoConfiguration.kt]: `[GoogleLLMClient]`

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * @param baseUrl Specifies the base URL for the LLM client.
+ * @param retry Defines the retry configuration for the LLM client using [RetryConfigKoogProperties].
+ */
+public interface KoogLlmClientProperties {
```
Comment (devcrocod) [inline koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt:15]: why not sealed interface?

**Review by devcrocod — APPROVED** [review]
lgtm 👍 

there are a few minor things

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt (hunk 4 above)
Comment (kpavlov) [reply koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/KoogLlmClientProperties.kt:15]: Sealed interface is not open for extension. It is very important for the libraries, because users might want to create their own starters for custom LLM clients.

File: koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/anthropic/AnthropicKoogProperties.kt
Comment (kpavlov) [reply koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/prompt/executor/clients/anthropic/AnthropicKoogProperties.kt:1]: I followed the package naming for the main classes.

There is no official naming convention, but probably the right package name should be **within or near the feature namespace** rather than in a low-context directory like `ai.koog.spring`.
E.g., for OpenRouter configuration, the package should be `ai.koog.prompt.executor.clients.openrouter.spring`
This naming convention requires individual starters for every LLM provider. It is actually the right thing to do because it allows users to choose the starters they need for their specific use case. I don't want to introduce individual starters in this PR, but the class structure is now more suitable for such refactoring.
